.changeset/silver-bears-wave.md

```diff
@@ -0,0 +1,5 @@
+---
+'@backstage/plugin-kubernetes-backend': patch
+---
+
+Fixed bug at Kubernetes proxy, now when ServiceAccount Strategy is used and no serviceAccountToken has been provided, the Kubernetes Proxy assumes backstage is running on k8s, so it is able to get the token and CA from the Pod instance
```

Comment:
```suggestion
Fixed a bug in the proxy endpoint. Now when the `serviceAccount` strategy is used and no `serviceAccountToken` has been provided, the proxy endpoint assumes backstage is running on Kubernetes and gets the URL and CA from the Pod instance.
```

Since getting the _token_ is already handled by the `ServiceAccountStrategy`:

https://github.com/backstage/backstage/blob/8cc5dd9d299ecc974c9f2b1dd9dc5ce0c6ab7a3d/plugins/kubernetes-backend/src/auth/ServiceAccountStrategy.ts#L38-L46